### ISSUE TITLE
Group membership not recognized with LDAP authentication

### ISSUE THREAD

**Issue by iamgrimm** [splitbrain/dokuwiki#3428]
I have to use LDAP so that I can route the traffic to a Duo Proxy for MFA. I can authenticate just fine with LDAP however I need to be able to use the groups the user is assigned to as a way to assign ACLs within dokuwiki. The groups do not appear to do anything within dokuwiki. I append &do=check and it tells me "You are part of the groups user". How do I get dokuwiki to recognize group membership? My local.protected.php filters are as follows:
```
$conf['plugin']['authldap']['server'] = 'ldap://domain.corp:389';
$conf['plugin']['authldap']['usertree'] = 'OU=Users,OU=project,DC=domain,DC=corp';
$conf['plugin']['authldap']['grouptree'] = 'ou=DokuWiki, ou=Groups, ou=project, dc=domain, dc=corp';
$conf['plugin']['authldap']['userfilter'] = '(&(sAMAccountName=%{user}))';
$conf['plugin']['authldap']['groupfilter'] = '(&(objectClass=*)(memberUID=%{user}))';
$conf['plugin']['authldap']['version'] = 3;
$conf['plugin']['authldap']['binddn'] = 'CN=dokusvcacct,OU=DokuWiki,OU=Service Accounts,OU=project,DC=domain,DC=corp';
$conf['plugin']['authldap']['bindpw'] = 'password';
$conf['plugin']['authldap']['attributes'] = array('cn', 'displayname', 'mail', 'givenname', 'objectclass', 'sn', 'uid', 'memberof');
```

**Comment by splitbrain** [comment]
Well it seems like your group filter or group tree is not correct. Since we don't know about the structure of your LDAP server we cannot tell you how to configure the plugin.

**Comment by iamgrimm** [comment]
> Well it seems like your group filter or group tree is not correct. Since we don't know about the structure of your LDAP server we cannot tell you how to configure the plugin.

Can you give me guidance?  I have been combing various forum posts for two days now trying every possible combination in other user posts, and nothing is working.  the grouptree points to the location of my groups, and I have dokuwikiuser, dokuwikiadmin, and dokuwiki_department1, dokuwiki_department2 groups within that OU.  How should I format the groupfilter to properly read those groups?  I am happy to provide the info you need for context.

**Comment by Klap-in** [comment]
There’s a `debug` setting, if you enable this you could get more information about what ldapauth is doing. That might be helpful as well.

**Comment by iamgrimm** [comment]
> There’s a `debug` setting for this plugin, if you enable this you could get more information about what ldapauth plugin is doing. That might be helpful as well.

I have it enabled, but it doesn't give me a ton of info.  I just disabled group filtering altogether by commenting the line out.  I tested service account I am binding with ability to get the attributes cn, displayname, mail, givenname, objectclass, sn, uid, and memberof by using ldapsearch `ldapsearch -x -b "dc=domain,dc=corp" -H ldap://domain.corp:389 -D "CN=dokusvcacct,OU=DokuWiki,OU=Service Accounts,OU=project,DC=domain,DC=corp" -W "sAMAccountName=test.user" cn displayname mail givenname objectclass sn uid memberof` and I got all the info back fine so the bind account is working properly.  I don't care about group filtering honestly, I can just let any user authenticate then control access with ACLs but I need to be able to use the memberof groups to build the ACLs around and when I create ACLs for specific groups they don't seem to work. I have 3 test pages, one viewable to all users, and the other two are accessible only to @dokuwikiadmin and @dokuwiki_department1. LDAP authentication occurs, and should pull the groups in based on my testing of memberof, but the ACLs do not apply.  I can not view the pages despite being in those groups in AD so DokuWiki is not seeing those groups despite them being readable by the bind account.
